### ISSUE TITLE
Bug: Not created groups from duplicated or imported allocation

### ISSUE THREAD

**Issue by alina-kiz** [learnweb/moodle-mod_ratingallocate#308]
1. Duplicate or import ratingallocation
2. Configure ratingallocate instance and choices
3. Create groups from allocation

Result Notification:
`The corresponding Moodle groups and groupings have been created. `
But groups is not created. 

The same is after all students allocations and publication.

**Comment by alina-kiz** [comment]
At importing or duplicating are groupingid = 0 and groupid = 0 stored in database tables ratingallocate_groupings, ratingallocate_group_choices, ratingallocate_ch_gengroups

**Comment by dlmsr** [comment]
Hi @alina-kiz , unfortunately I cannot reproduce this behaviour with Moodle 4.4.6 (Build: 20250210). What Moodle version do you use?

**Comment by alina-kiz** [comment]
Hi @dlmsr,  Moodle 4.1.16 (Build: 20250210), mod_ratingallocate v4.4-r2 2024082600

**Comment by dlmsr** [comment]
I tested it with the versions you specified now and I still could not reproduce the bug. Could you be more specific with the steps you take? How do you setup the ratingallocate instance? Are there any students in the course? I have done the following and did not observe the described behaviour:

1. Create a new course
2. Create a ratingallocate instance
3. Add one choice
4. Duplicate the activity
5. Open the duplicated activity
6. Click on "Create groups from allocations" button

The respective group was created.

**Comment by alina-kiz** [comment]
Hi, the steps:
1. Create a new course
2. Create a ratingallocate instance
3. Add a choices
4. Wait until Last algorithm run at
5. Click Distribute equally
6. Wait until distribute finished
7. Click Create Groups from Allocation
8. Create another new course
9. Go to Course Reuse
10. Import first course into second course, include activities, resources, blocks, files, filters, question bank, custom fields, bank content, course files with ratingallocate instance
11. Edit imported ratingallocate
12. Edit choices
13. Wait until Last algorithm run at
14. Click Distribute equally
15. Wait until distribute finished
16. Click Create Groups from Allocation
17. Go to course participants -> Groups

Groups from Allocation are not created

or 

12. Add choices
13. Wait until Last algorithm run at
14. Click Distribute equally
15. Wait until distribute finished
16. Click Create Groups from Allocation

Error:
Can't find data record in database table course.
More information about this error

**Comment by alina-kiz** [comment]
Yes, there were also students enrolled in the courses and divided into groups.

**Comment by dlmsr** [comment]
I could reproduce this bug using Ratingallocate v4.5-r1. The following steps were necessary:

- In a ratingallocate activity in a course T1, click _Create Groups from Allocation_.
- Make a course backup of this course without including groups or groupings.
- Create a course T2 and restore from the previous backup

In this new course T2, clicking _Create Groups from Allocation_ it says _The corresponding Moodle groups and groupings have been created_, however, this is not the case. This bug is fixed by #317, therefore I am closing this issue.
